### PR TITLE
Make CUDA triu / tril support batches of size > 65535

### PR DESCRIPTION
**Pull request by vishwakftw**
In the previous implementation of triu / tril, we passed the batch size in the 2nd dimension of a grid. This is limited to 65535, which means that performing triu / tril on a tensor with batch size > 65535 will throw an error. This PR removes the dependence on the 2nd dimension, and corresponding non-contiguity constraints.

Changelog:
- Compute offset, row and col in the kernel
- Use 1st dimension of grid alone
- Remove unnecessary contiguity checks on tensors as a result of this change.

Test Plan:
- All tests should pass to verify that the change is correct.

### CODE REVIEW

File: aten/src/ATen/native/cuda/BatchLinearAlgebra.cu [hunk 1]
```diff
@@ -801,53 +802,60 @@ C10_LAUNCH_BOUNDS_1(512)
 #endif
 __global__
 void triu_tril_kernel(
-    scalar_t* result, scalar_t* self, int64_t k, int64_t N,
-    int64_t res_batch_stride, int64_t res_row_stride, int64_t res_col_stride,
-    int64_t self_batch_stride, int64_t self_row_stride, int64_t self_col_stride, int64_t self_ncol) {
+    cuda::detail::TensorInfo<scalar_t, int64_t> result_info,
```
Comment (ngimel) [inline aten/src/ATen/native/cuda/BatchLinearAlgebra.cu]: Also, if performance is important, then templating on index type and using int32_t for smaller tensors helps a lot.

**Review by ngimel — COMMENTED** [review]
I'm ok with this PR if we need only functional support, if performance is important some changes are necessary. Code is nicely simplified.

File: aten/src/ATen/native/cuda/BatchLinearAlgebra.cu (hunk 1 above)
Comment (vishwakftw) [reply aten/src/ATen/native/cuda/BatchLinearAlgebra.cu]: Thank you for the comment. In this case, I would have to check if 32 bit indexing can be used, and then use the appropriate index type, correct?

File: aten/src/ATen/native/cuda/BatchLinearAlgebra.cu [hunk 2]
```diff
@@ -801,53 +802,60 @@ C10_LAUNCH_BOUNDS_1(512)
...
   }
 
-  int64_t self_batch_idx = blockIdx.y;
-  int64_t row = linear_idx / self_ncol;
-  int64_t col = linear_idx % self_ncol;
+  auto dims = self_info.dims;
+
+  int64_t self_offset = 0, result_offset = 0;
+  // Compute column index and corresponding offset
+  int64_t col = linear_idx % self_info.sizes[dims - 1];
```
[the remark this answers is not in the document]
Comment (vishwakftw) [reply aten/src/ATen/native/cuda/BatchLinearAlgebra.cu]: Sure, thank you.

**Comment by vishwakftw** [comment]
@ngimel This is not a necessarily performance critical kernel - it's generally used in tandem with many linear algebra operations. However, since you mentioned that indexing based on int32 helps a lot for smaller inputs, I've incorporated it in the PR.

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**